### DISCUSSION TITLE
Why `<leader>` (and `<tab>`) is lowercased, but other special keys (`<Up>`, `<Down>`, `<Esc>`, `<C-...>`) are not?

### DISCUSSION THREAD

**Discussion by Andrew15-5** [NvChad/NvChad#2480] [Q&A] [answered]
As a perfectionist, I find it very <strike>annoying</strike>strange that only `<leader>` key is lowercased, but all others are not. This is very important when disabling existing mappings, as you can't disable `["<leader>abc"]` with `["<Leader>abc"]`, because these strings are not identical/equal. Related to this question is that all English letters are also lowercased.

The Neovim help doc shows examples of mappings in capitalized form (`h <C-`): `<S-Right>`, `<M-C-T>`. The is also `h <Leader>`.

Can you justify this inconsistency, and will it be possible to make everything capitalized (first letters in uppercase) or everything lowercased (not my preferred way as of right now, but it is definitely easier to type out)?

Edit1: Oh, right, `<tab>` is also lowercased for some reason.

Edit2: I found even more inconsistency:

https://github.com/NvChad/NvChad/blob/a8413849cf9d648df0ae81a9349c8b125e628919/lua/core/mappings.lua#L121

Now there is `<Esc>` and `<ESC>` in the project. And `<cmd>` is also lowercased.

**Comment by siduck** [comment, marked as answer]
ok will make em all lowercased

**Reply by Andrew15-5** [reply]
This will probably be released under version 3, since it will break `M.disabled` part of the custom config of the users, right? Assuming [semver](https://semver.org/) is used.

**Reply by siduck** [reply]
yes v3.0
